### PR TITLE
Add tests for packet_alert_max - v4

### PR DESCRIPTION
**Pull request by jufajardini**
Previous PR: https://github.com/OISF/suricata-verify/pull/743

Changes:
- Add test for corner case situation described by @catenacyber in https://github.com/OISF/suricata/pull/6896#discussion_r796064516

Related to https://redmine.openinfosecfoundation.org/issues/4207
and https://redmine.openinfosecfoundation.org/issues/4942

Goes with Suricata PR

### CODE REVIEW

File: tests/alert-max/alert-max-append-higher-priority/test.rules [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# internal id 0 -- should not trigger
```
Comment (catenacyber) [inline tests/alert-max/alert-max-append-higher-priority/test.rules:1]: This looks complex... Will it be stable ?

Comment (jufajardini) [reply tests/alert-max/alert-max-append-higher-priority/test.rules:1]: Not sure I am the best one to judge, but Victor agreed that we should try to have this test as an S-V one...

File: tests/alert-max/alert-max-append-higher-priority/suricata.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      enabled: yes
+      filetype: regular #regular|syslog|unix_dgram|unix_stream|redis
+      filename: eve.json
+      types:
+        - alert
+        - http
+
+# Define maximum number of possible alerts that can be triggered for the same
+# packet. Default is 15
+packet-alert-max: 2
```
Comment (jufajardini) [inline tests/alert-max/alert-max-append-higher-priority/suricata.yaml:21]: packet-alert-max had to be `2` instead of the otherwise obvious 3, because some of the rules are inspected against a fake packet, apparently, leading to a packet-alert-max value = 3 allowing for 4 alerts to be queued.

File: tests/alert-max/alert-max-append-higher-priority/test.rules (hunk 1 above)
Comment (catenacyber) [reply tests/alert-max/alert-max-append-higher-priority/test.rules:1]: Good enough for me :-)

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata-verify/pull/808
